src/ast/base_types.h

```diff
@@ -4,19 +4,31 @@ namespace ast {
 
 class Identifier {
  public:
-  explicit Identifier(std::string name) : name_(std::move(name)) {}
+  explicit Identifier(std::string name, lexer::Range location,
+                      bool is_uppercase, bool absolute = false)
+      : name_(std::move(name)),
+        location_(std::move(location)),
+        absolute_(absolute),
+        is_uppercase_(is_uppercase) {}
   const std::string& to_string() const { return qualified_name(); }
```

Comment:
Better to move the review to reviewable.io, but this is just to provide a common interface to all printable types.